### ISSUE TITLE
`zero-stepsize-in-slice` - false negative in the example from the docs

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ty#3736]
### Summary

https://docs.astral.sh/ty/reference/rules/#zero-stepsize-in-slice

```py
l = list(range(10))
l[1:10:0] # no error
```

[Playground](https://play.ty.dev/323aa04e-7ae1-4426-ad22-33fbc6f9949b)

### Version

267af002f

**Comment by MichaReiser** [comment]
Thanks. 

This seems intentional, but we should update our documentation

> We raise a `zero-stepsize-in-slice` diagnostic when trying to slice a literal string, bytes, or
tuple with a step size of zero (see tests in `string.md`, `bytes.md` and `tuple.md`). But we don't
want to raise this diagnostic when slicing a custom type:

**Comment by DetachHead** [comment]
why not also report it on `list` though? it's not a custom type

**Comment by MichaReiser** [comment]
Because subclassing https://github.com/astral-sh/ruff/pull/13917#discussion_r1819996605. I'm not sure if we should reconsider, given that we have changed our stand point on some other subclassing cases too

**Comment by carljm** [comment]
Hmm I think I was kind of wrong there tbh. Even though a hypothetical subclass might be safe, we should still error if the type is `list`, because that type definitely includes objects of type `list` which are known to not be safe with step size zero. So I think we should ideally fix this in the implementation (though if that's tricky we could also update the docs in the short term, and we should at least do that before stable release.)

**Comment by charliermarsh** [comment]
Hopefully interpreted the ask correctly: https://github.com/astral-sh/ruff/pull/25966
